url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCHandPanel.java

```diff
@@ -0,0 +1,3030 @@
+            doLayout();  // must move offer panel
+            repaint();
+        }
+    }
+
+    /**
+     * Clear the current offer.
+     * If player is client, clear the numbers in the resource "offer" squares,
+     * and disable the "offer" and "clear" buttons (since no resources are selected).
+     * Otherwise just hide the last-displayed offer.
+     *
+     * @param updateSendCheckboxes If true, and player is client, update the
+     *    selection checkboxes for which opponents are sent the offer.
+     *    If it's currently our turn, check all boxes where the seat isn't empty.
+     *    Otherwise, check only the box for the opponent whose turn it is.
+     */
+    public void clearOffer(boolean updateSendCheckboxes)
+    {
+        if (! offerIsResetMessage)
+        {
+            offer.setVisible(false);
+            offer.clearOffer();  // Clear to zero the offer and counter-offer
+        }
+
+        if (playerIsClient)
+        {
+            // clear the squares panel
+            sqPanel.setValues(zero, zero);
+
+            // reset the send squares (checkboxes)
+            if (updateSendCheckboxes && ! playerTradingDisabled)
+            {
+                int pcurr = game.getCurrentPlayerNumber();  // current player number
+                boolean pIsCurr = (pcurr == player.getPlayerNumber());  // are we current? 
+                for (int i = 0; i < game.maxPlayers - 1; i++)
+                {
+                    boolean canSend;
+                    if (pIsCurr)
+                        // send to any occupied seat
+                        canSend = ! game.isSeatVacant(playerSendMap[i]);
+                    else
+                        // send only to current player
+                        canSend = (pcurr == playerSendMap[i]);
+                    playerSend[i].setBoolValue(canSend);
+                    playerSend[i].setEnabled(canSend);
+                }
+            }
+
+            clearOfferBut.disable();
+            if (! playerTradingDisabled)
+            {
+                offerBut.disable();
+                offerButTip.setTip(OFFERBUTTIP_DIS);
+            }
+        }
+        else if (offerHidesControls)
+        {
+            hideTradeMsgShowOthers(true);
+        }
+        validate();
+        repaint();
+    }
+
+    /**
+     * Show or hide a message in the trade-panel.
+     * Should not be client player, only other players.
+     * Sets offerIsMessageWasTrade, but does not set boolean modes (offerIsResetMessage, offerIsDiscardMessage, etc.)
+     * Will clear boolean modes if message null.
+     *
+     * @param message Message to show, or null to hide (and return tradepanel to previous display, if any)
+     */
+    private void tradeSetMessage(String message)
+    {
+        if (playerIsClient)
+            return;
+        if (message != null)
+        {
+            offerIsMessageWasTrade = (offer.isVisible() && (offer.getMode() == TradeOfferPanel.OFFER_MODE));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.